### ISSUE TITLE
[question] What magnitude of training data do we need to use AgentFlow?

### ISSUE THREAD

**Issue by vanetreg** [lupantech/AgentFlow#19]
I processed your repo, docs and Deepwiki pages in NotebookLM and I was told we need 10k+ data points to train the system properly.
Is this true?
This can be a huge barrier for many of us, 
so pls. clarify, when and under what use cases and circumstances it is / it is not true!
Thank you! :)

**Comment by IsaacGHX** [comment]
Hi @vanetreg 

Thank you for your question! The amount of training data needed depends significantly on your training configuration and objectives.

## Data Requirements Based on Training Configuration

**If you want all problems seen during training to be one-shot (seen only once)**, the calculation is straightforward:

```
Required Data = batch_size × training_steps
```

For example, with `batch_size = 32` and `training_steps = 100`:
```
Required Data = 32 × 100 = 3,200 samples
```

So you would only need **~3.2K data points** for this configuration, which is much more accessible than 10K+.

## Recommended Data Composition Strategy

Given our shuffled training approach, we recommend **sampling equal amounts of diverse, discriminative problems from different data sources**. For instance:

- **50% Math problems** (e.g., [DeepMath-103K](https://huggingface.co/datasets/zwhe99/DeepMath-103K))
- **50% Search-based problems** (e.g., [NQ (Natural Questions)](https://huggingface.co/datasets/RUC-NLPIR/FlashRAG_datasets))

This balanced approach ensures the planner learns to handle different task types and tool requirements effectively.

## Key Considerations

1. **Data diversity matters more than quantity** - Having problems that require different reasoning patterns and tools is crucial
2. **The 10K+ figure** may apply if you're doing multiple epochs over the data or need extensive coverage of edge cases
3. **Start small** - You can begin with 1K well-curated samples and scale up based on performance

The barrier to entry is much lower than 10K+ samples if you configure your training appropriately!

Feel free to ask if you need help with data preparation or configuration settings.
